### ISSUE TITLE
[BUG] cannot schedule cattle-cluster-agent on tainted EKS nodes

### ISSUE THREAD

**Issue by zbialik** [rancher/rancher#40418]
**Rancher Server Setup**
- Rancher version: `v2.6.8`
- Installation option (Docker install/Helm Chart): Helm (run `helm template` to generate manifests for a GitOps deployment)
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): `v2.6.8`
- Proxy/Cert Details: we use an AWS ALB with ACM (works fine)

**Information about the Cluster**
- Kubernetes version: `v1.24.9`
- Cluster Type (Local/Downstream): Imported EKS clusters in Rancher
- **Worker Nodes are ALL tainted**

**Describe the bug**

The `cattle-cluster-agent` `Deployment` can't schedule because it doesn't `tolerate` our custom `taints` on the EKS Worker Nodes

**To Reproduce**

1. Create an EKS Cluster with a NodeGroup that is configured with a custom `taint`
2. Initialize the cluster in Rancher UI
3. Run the `kubectl apply` command for importing the cluster to Rancher

**Result**

The `cattle-cluster-agent` `Deployment` can't schedule because it doesn't `tolerate` our custom `taints` on the EKS Worker Nodes

**Expected Result**

I expect to be able to configure the `Deployment` with a custom `toleration` **_somehow_**. 

**Additional context**

If you try to customize the provided `Deployment` manifest with the needed `toleration` before running the `kubectl apply`, Rancher will still patch it back to the original value as per the new cluster registration process. 

Related Issues:
- https://github.com/rancher/rancher/issues/24636 (would have worked if this wasn't implemented)
- https://github.com/rancher/rancher/issues/36788 (shouldn't have been closed)
- https://github.com/rancher/rancher/issues/35834 (similar issue for fleet-agent)

**Comment by zbialik** [comment]
I think the real solution is for https://github.com/rancher/rancher/issues/39711 to be implemented.

**Comment by zbialik** [comment]
@superseb raising this to you

this prevents us from implementing an important multi-tenant cluster design where we have custom `taints` on **all** our `Nodes` in an EKS cluster.

Since https://github.com/rancher/rancher/pull/30304 only applies `tolerations` based on defaults set for EKS, we have no way of adding our needed custom `tolerations` to the workload.

**Comment by superseb** [comment]
The implemented logic seems to be the same as when I created it, and it will only look for nodes with control plane labels. This will not work in case of custom taints/tolerations, and needs additional work to discuss what to implement to fulfil most needs.

**Comment by zbialik** [comment]
Yeah I mainly pointing out that the newly introduced logic prevents users from import k8s clusters that leverage custom `taints`/`tolerations` for all `Nodes`.

One suggestion I have is to provide users a custom `label` (maybe even re-use the `cattle.io/cluster-agent=true` label) that can tell the agent the `taints` to tolerate. This is already being done for `Nodes` that the agent finds with the  _controlplane_ `labels`. Could this logic be extended to another `label` like `cattle.io/cluster-agent=true`?

**Comment by superseb** [comment]
@zbialik I guess https://github.com/rancher/rancher/issues/41035 should fulfil your needs, please see if it does and provide feedback on the implementation.

**Comment by slickwarren** [comment]
we have this in the test plan for the ticket mentioned above
